R/hazard_link.R

```diff
@@ -20,34 +20,37 @@
 #' @export
 HazardLink <- setClass(
     "HazardLink",
-    representation = list(
-        "stan" = "StanModule",
-        "contribution" = "character",
-        "parameters" = "character"
+    slots = list(
+        stan = "StanModule",
+        contribution = "character",
+        parameters = "character"
     )
 )
 
+setValidity("HazardLink", function(object) {
+    if (!length(object@parameters) >= 1) {
+        "`parameter` must be a vector"
+    } else {
+        TRUE
+    }
+})
+
+setValidity("HazardLink", function(object) {
+    if (!length(object@contribution) == 1) {
+        "`contribution` must be length 1 character vectors"
+    } else {
+        TRUE
+    }
+})
+
```

Comment:
I think these overwrite each other,  I don't think they append for example:

```
myc <- setClass("myc", slots = list(x = "character"))

setValidity("myc", function(object) length(object@x) > 2)
setValidity("myc", function(object) length(object@x) == 1)

x <- myc(x = "a")
```

Doesn't error because the second condition takes precedence